src/Stiefel.jl

```diff
@@ -0,0 +1,239 @@
+    end
+    if (T <: Complex) && !(eltype(x) <: Real) && !(eltype(x) <: Complex)
+        return DomainError(eltype(x),
+            "The matrix $(x) is neiter real- nor complex-valued matrix, so it does noe lie on the complex Stiefel manifold of dimension ($(M),$(N)).")
+    end
+    if any( size(x) != representation_size(S) )
+        return DomainError(size(x),
+            "The matrix $(x) is does not lie on the Stiefel manifold of dimension ($(M),$(N)), since its dimensions are wrong.")
+    end
+    c = x'*x
+    if !isapprox(c, one(c); kwargs...)
+        return DomainError(norm(c-one(c)),
+            "The point $(x) does not lie on the Stiefel manifold of dimension ($(M),$(N)), because x'x is not the unit matrix.")
+    end
+end
+
+function check_tangent_vector(S::Stiefel{M,N,T},x,v; kwargs...) where {M,N,T}
+    t = check_manifold_point(S,x)
+    if (t != nothing)
+        return t
+    end
+    if (T <: Real) && !(eltype(v) <: Real)
+        return DomainError(eltype(v),
+            "The matrix $(v) is not a real-valued matrix, so it can not be a tangent vector to the Stiefel manifold of dimension ($(M),$(N)).")
+    end
+    if (T <: Complex) && !(eltype(v) <: Real) && !(eltype(v) <: Complex)
+        return DomainError(eltype(v),
+            "The matrix $(v) is neiter real- nor complex-valued matrix, so it can not bea tangent vectorto the complex Stiefel manifold of dimension ($(M),$(N)).")
+    end
+    if any( size(v) != representation_size(S) )
+        return DomainError(size(v),
+            "The matrix $(v) is does not lie in the tangent space of $(x) on the Stiefel manifold of dimension ($(M),$(N)), since its dimensions are wrong.")
+    end
+    if !isapprox(x'*v + v'*x, zeros(N,N); kwargs...)
+        return DomainError(norm(x'*v + v'*x),
+            "The matrix $(v) is does not lie in the tangent space of $(x) on the Stiefel manifold of dimension ($(M),$(N)), since x'v + v'x is not the zero matrix.")
+    end
+end
+@doc doc"""
+    exp!(M,y, x, v)
+
+compute the exponential map on the [`Stiefel`](@ref)`{m,n,T}`() manifold `M`
+eminating from `x` into tangent direction `v` and store the result in `y`.
+The formula reads
+
+$\operatorname{exp}_{x} v = \begin{pmatrix}
+   x\\v
+ \end{pmatrix}
+ \exp
+ \left(
+ \begin{pmatrix} {\bar x}^{\mathrm{T}}v & -{\bar \xi}^{\mathrm{T}}v\\
+ I_n & {\bar x}^{\mathrm{T}}v\end{pmatrix}
+ \right)
+\begin{pmatrix}  \exp( -{\bar x}^{\mathrm{T}}v) \\ 0_n\end{pmatrix}$
+
+where $\exp$ denotes matrix exponential, and $I_n$ and
+$0_n$ are the identity matrix and the zero matrix of dimension $n \times n$,
+respectively.
+"""
+function exp!(M::Stiefel{m,n,T}, y, x, v) where {m,n,T}
+    y .= [x v] * exp([x'v   -v'*v; one(zeros(T,n,n))   x'*v]) * [exp(-x'v); zeros(T,n,n)]
+    return y
+end
+
+@doc doc"""
+    inner(M,x,ξ,ν)
+
+compute the inner product for two tangent vectors `v`, `w` from the
+tangent space of `x` on the [`Stiefel`](@ref) manifold `M`. The formula reads
+
+````math
```

Comment:
```suggestion
```math
```

Reply:
Oh, hm, no. You can start code with any number of back tricks, and this one starts and ends with 4, as do all in that file. Should we default to 3? I don't have a strong preference, just that I have sometimes seen 3 to indicate something different (don't remember what, but that's when I switched to 4).

Reply:
Hm, my misunderstanding. For some reason, this equation [isn't rendering]( https://julianlsolvers.github.io/Manifolds.jl/previews/PR49/manifolds/stiefel.html#ManifoldsBase.inner-Union{Tuple{T},%20Tuple{N},%20Tuple{M},%20Tuple{Stiefel{M,N,T},Any,Any,Any}}%20where%20T%20where%20N%20where%20M)

Reply:
Ah, spaces upfront, especially at the closing ticks.